aca_preview/preview.py

```diff
@@ -457,71 +460,81 @@ def check_enough_guide_for_ers(self, n_required=8):
+            (as if they were in one pixel).  This is somewhat the inverse of
+            proseco.get_pixmag_for_offset .
+            """
+            cand_counts = mag_to_count_rate(cand_mag)
+            spoil_counts = mag_to_count_rate(imposter_mag)
+            return spoil_counts * 3 * 5 / (spoil_counts + cand_counts)
+
+        agasc_id = star['id']
+        idx = self.get_id(agasc_id)['idx']
+        offset = imposter_offset(star['mag'], star['imp_mag'])
+        for limit, category in ((4.0, 'critical'),
+                                (2.5, 'warning')):
+            if offset > limit:
+                self.add_message(
+                    category,
+                    f'Guide star imposter offset {offset:.1f}, limit {limit} arcsec',
+                    idx=idx)
+                break
 
-    def check_too_bright_guide(self):
+    def check_too_bright_guide(self, star):
         """Warn on guide stars that may be too bright.
 
         - Critical if MAG_ACA_ERR used in selection is less than 0.1
-        - Critical if within 3 * mag_err of the hard 5.8 limit.
+        - Critical if within 2 * mag_err of the hard 5.8 limit, warn within 3 * mag_err
         - Warning if brighter than 6.1 (should be double-checked in
           context of other candidates).
 
         """
-        for star in self.guides:
-            agasc_id = star['id']
-            if star['mag'] - (3 * star['mag_err']) < 5.8:
-                self.add_message(
-                    'critical',
-                    f'Guide star {agasc_id} within 3*mag_err of 5.8')
-            elif (star['mag'] < 6.1) and (star['MAG_ACA_ERR'] * 0.01 < 0.1):
-                self.add_message(
-                    'critical',
-                    f'Guide star {agasc_id} < 6.1 with small MAG_ACA_ERR.  Double check selection.')
-            elif star['mag'] < 6.1:
+        agasc_id = star['id']
+        idx = self.get_id(agasc_id)['idx']
+        mag_err = star['mag_err']
+        mag_aca_err = star['MAG_ACA_ERR'] * 0.01
+        for mult, category in ((2, 'critical'),
+                               (3, 'warning')):
+            if star['mag'] - (mult * mag_err) < 5.8:
                 self.add_message(
-                    'warning',
-                    f'Guide star {agasc_id} < 6.1. Double check selection.')
+                    category,
+                    f'Guide star {agasc_id} within {mult}*mag_err of 5.8 '
+                    f'(mag_err={mag_err:.2f})', idx=idx)
+                break
+        if (star['mag'] < 6.1) and (mag_aca_err < 0.1):
+            self.add_message(
+                'critical',
+                f'Guide star {agasc_id} < 6.1 with small MAG_ACA_ERR={mag_aca_err:.2f}.  '
+                f'Double check selection.',
+                idx=idx)
+        elif star['mag'] < 6.1:
+            self.add_message(
+                'warning',
+                f'Guide star {agasc_id} < 6.1. Double check selection.', idx=idx)
```

Comment:
It is likely that a star that hits the mult*mag_err will also hit this one, but I'm OK with one critical warning and one plain warning in that case.

Reply:
I had the same thought and the same "I'm OK" for now.  I hope to make these tests tighter with experience (and more time to think about it).